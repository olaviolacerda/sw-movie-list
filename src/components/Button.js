import React from 'react';
import PropTypes from 'prop-types';

export default function Button({className, children, handleClick}) {
  return (
    <button className={className} type="button" onClick={handleClick}>
      {children}
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  handleClick: PropTypes.func,
  children: PropTypes.node,
};

Button.defaultProps = {
  className: '',
  children: 'Click me',
  handleClick: () => console.log('clicked'),
};
