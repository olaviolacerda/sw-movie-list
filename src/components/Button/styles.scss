.button {
  width: 150px;
  height: 50px;
  background: #016cf9;
  border: 0;
  border-radius: 50px;
  border-color: #016cf9;
  align-self: center;
  cursor: pointer;
  margin-top: 10px;
  position: absolute;
  bottom: 20px;

  a {
    color: #fff;
    text-decoration: none;
    text-transform: uppercase;
    font-weight: 500;
  }
}
