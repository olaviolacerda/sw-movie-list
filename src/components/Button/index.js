import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

export default function Button({className, children, handleClick}) {
  return (
    <button className={className} type="button" onClick={handleClick}>
      {children || null}
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  handleClick: PropTypes.func,
  children: PropTypes.node,
};

Button.defaultProps = {
  className: 'button',
  handleClick: () => {},
  children: null,
};
