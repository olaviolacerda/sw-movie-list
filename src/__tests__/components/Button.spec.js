import React from 'react';
import {shallow} from 'enzyme';
import Button from '../../components/Button';

describe('Component Button test', () => {
  it('should render without crashing', () => {
    shallow(<Button />);

    expect(Button).toBeDefined();
  });

  it('should render correctly', () => {
    const buttonElement = shallow(
      <Button className="button" onClick={() => {}} type="button">
        button test
      </Button>
    );

    expect(buttonElement).toMatchSnapshot();
  });

  it('should have a button value', () => {
    const buttonElement = shallow(
      <Button className="button" onClick={() => {}} type="button">
        button test
      </Button>
    );
    expect(buttonElement.props().children).toEqual('button test');
  });
});
