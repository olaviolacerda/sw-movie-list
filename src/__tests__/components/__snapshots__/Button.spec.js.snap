// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button test should render correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  button test
</button>
`;
